### PR TITLE
beast: configure beagle dependency

### PR DESCRIPTION
**Pull request by suujia**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Linuxbrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Linuxbrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
```
Failed to load BEAGLE library: no hmsbeagle-jni in java.library.path
```

### CODE REVIEW

**Comment by suujia** [comment]
```

$brew uses beast 
Error: Failed to import: /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/llvm@4.rb
Error: Failed to import: /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/android-platform-tools.rb

glibc: N/A
Error: uninitialized constant Formulary::FormulaNamespaceb01da2f4c338ba50fbf392629843ff26::Gcc::GlibcRequirement
```

**Comment by sjackman** [comment]
Please try rebasing onto `origin/master` and adding `revision 1`.
```sh
git checkout master
git pull --ff-only
git checkout beast
git rebase origin/master
git push suujia beast --force
```

**Comment by suujia** [comment]
Thanks! 
```
==> brew bottle --merge --write --no-commit ./beast--1.10.1_1.x86_64_linux.bottle.json --keep-old
Error: --keep-old was passed but there are changes in:
sha256 => x86_64_linux
Error: Kernel.exit

Picked up _JAVA_OPTIONS:  -Duser.home=/home/linuxbrew/.cache/Homebrew/java_cache
Error: Could not find or load main class dr.app.beast.BeastMain
Caused by: java.lang.ClassNotFoundException: dr.app.beast.BeastMain
Error: beast: failed
```

**Comment by sjackman** [comment]
I forget that Linuxbrew/core is a bit different.
Change `revision 1` to `revision 1 unless OS.mac?`
Remove the line `sha256 "…" => :x86_64_linux`

The `brew test beast` failure you'll need to troubleshoot and debug locally.

File: Formula/beast.rb [hunk 1]
```diff
@@ -31,6 +31,7 @@ def install
     libexec.install Dir["release/Linux/BEASTv*/*"]
     pkgshare.install_symlink libexec/"examples"
     bin.install_symlink Dir[libexec/"bin/*"]
+    inreplace bin/"beast", "/usr/local/lib:$LD_LIBRARY_PATH", Formula["beagle"].lib
```
Comment (sjackman) [inline Formula/beast.rb]: `bin/beast` is a symlink, but `inreplace` is replacing it with a file, which won't work. Try…
```ruby
inreplace libexec/"bin/beast", "/usr/local/lib:$LD_LIBRARY_PATH", Formula["beagle"].lib unless OS.mac?
```

**Comment by suujia** [comment]
@sjackman Perfect, it worked 😃

**Comment by sjackman** [comment]
I'm running into an error when pulling this bottle. Could you please try rebasing this PR onto `master` to rebuild the bottles?

**Comment by sjackman** [comment]
Merged. Thanks, Susie! 🏆

**Pull request closed without merging**